docs: Prometheus database snapshot (#3348)
scality/metalk8s: docs/operation/cluster_monitoring/prometheus.rst

.. csv-table:: Default Prometheus Alerting rules
   :file: ../../../tools/rule_extractor/alerting_rules.csv
   :header: "Name", "Severity", "Description"

Snapshot Prometheus Database
----------------------------

> Snapshotting Prometheus database can be useful for backup and restore,

Reviewer: If we mention backup/restore, we need to explain _how to restore_, which is not straightforward. I'd remove this part for now.

Author: I wasn't thinking of MetalK8s backup/restore mechanisms, it was more a backup/restore of the Prometheus database itself.

Author: I can remove it if you find it confusing.

Reviewer: > a backup/restore of the Prometheus database itself

Yes I got that, my point was that restoring from a snapshot is not obvious, so if we mention this, we need to at least give pointers for how to do it (it's not even part of the official Prometheus docs). But anyway, that's also true about "later investigations", a user would need to know how to _use_ the generated snapshots. Maybe in a later iteration of this ticket.

Author: I've removed this paragraph.

but also to export the platform metrics in order to investigate, for
example, a performance issue.

To snapshot the database, you must first
:ref:`enable the Prometheus admin API<csc-enable-prometheus-admin-api>`.

> You can then generate the snapshot using
> :ref:`sosreport utility<sosreport-generate-report>` with the following options:

Reviewer: ```suggestion
To generate a snapshot, use the
:ref:`sosreport utility<sosreport-generate-report>` with the following options:
```

.. code-block:: shell

>    sosreport --batch -o metalk8s -kprometheus-snapshot=True

Reviewer: OK, but where is the snapshot generated? Any concern about size?

Author: Added an explanation and a warning.
